### PR TITLE
Error on export default of type: issue https://github.com/microsoft/TypeScript/issues/55087

### PR DESCRIPTION
**Pull request by EliLichtblau**
Fixes: [#55087](https://github.com/microsoft/TypeScript/issues/55087)
This changes the behavior of TS to the specified behavior in [issue](https://github.com/microsoft/TypeScript/issues/55087).

Export default of types become errors - i.e
```ts
type foo = 3
export default foo
```
This is because if it does not error - typescript emits invalid code.

### CODE REVIEW

**Comment by EliLichtblau** [comment]
@microsoft-github-policy-service agree

**Comment by EliLichtblau** [comment]
Fixes: [#55087](https://github.com/microsoft/TypeScript/issues/55087)

- trying to trigger link issue

**Comment by fatcerberus** [comment]
FWIW I think it's okay to default-export types in general, as long as they *only* have a typespace meaning--the problem is when you default-export something that has a valuespace meaning (like a class) which has been imported via `import type`.

**Comment by EliLichtblau** [comment]
If this is the only condition - I think you can just convert 
```ts
 if (!isIllegalExportDefaultInCJS && !(node.flags & NodeFlags.Ambient) && compilerOptions.verbatimModuleSyntax && getTypeOnlyAliasDeclaration(sym, SymbolFlags.Value)) {
```
to 
```ts
if (!(node.flags & NodeFlags.Ambient) && getTypeOnlyAliasDeclaration(sym, SymbolFlags.Value)) {
```

the reason I find this a little awkward is checking with ts-fork-typechecker but emitting with ts loader transpileOnly is pretty common and you can pass the typechecking stage and still emit invalid code as the following code:
```ts
ts.transpileModule(`
import {type Node} from "typescript"
export default Node
`, {})
```
will give you
```ts
export default Node
```
even though Node is an interface. I can understand why you (the TS team) would say this is improper use and shouldn't be supported though.

With that said, my take is: 
* There should be a flag which disables all export default of things (need a better word here sorry) imported in a typeOnly context as this can emit invalid code with transpileModule calls
* The check on export default of import types should not live behind the verbatim module syntax flag and should be default behavior 


But I will implement whatever you think is best

The best solution IMO is to not change compile time behavior at all and for emit to just work but based on your comments in the issue I am unsure if this is possible. I know less than little about the emit phase so I don't have any idea here.

**Review by andrewbranch — CHANGES_REQUESTED** [review]
I think the expected behavior is

- an `export default` of a type-only import never emits anything
- it’s an error only in `isolatedModule` (ts1205)

**Comment by andrewbranch** [comment]
Fixed the checker behavior, still need to fix the emit

**Comment by andrewbranch** [comment]
Ok, ready.

File: tests/baselines/reference/exportDefault.js [hunk 1]
```diff
@@ -41,7 +41,6 @@ exports.A = A;
 //// [b.js]
 "use strict";
 Object.defineProperty(exports, "__esModule", { value: true });
-exports.default = types;
```
Comment (jakebailey) [inline tests/baselines/reference/exportDefault.js:44]: Oops!

**Review by jakebailey — APPROVED** [review]
LGTM but should get the extended test run.

**Comment by jakebailey** [comment]
@typescript-bot test top200
@typescript-bot user test this
@typescript-bot run dt

@typescript-bot perf test this

File: tests/baselines/reference/exportDefault.js (hunk 1 above)
Comment (andrewbranch) [reply tests/baselines/reference/exportDefault.js:44]: Yeppp

**Comment by andrewbranch** [comment]
These errors from the top repos look possibly wrong, especially the playwright one. Need to take a closer look.

**Comment by andrewbranch** [comment]
@typescript-bot test top200

**Comment by jakebailey** [comment]
You can ignore that playwright one, that is currently happening on every PR and we just haven't looked at why.

**Comment by andrewbranch** [comment]
Huh? The playwright one is clearly caused by this and fixed in the latest commit.

**Comment by jakebailey** [comment]
Apologies, I mixed up the "P" libraries and confused playwright for puppeteer (which you can see on other PRs like https://github.com/microsoft/TypeScript/pull/56429#issuecomment-1814887414). Ignore me!

**Pull request merged**